src/mongo/registerMongoCommands.ts

```diff
@@ -44,23 +48,24 @@ export function registerMongoCommands(context: vscode.ExtensionContext, actionHa
         if (!node) {
             node = <IAzureParentNode<MongoDatabaseTreeItem>>await tree.showNodePicker(MongoDatabaseTreeItem.contextValue);
         }
-        if (connectedDb) {
-            connectedDb.treeItem.isConnected = false;
-            await connectedDb.refresh();
-        }
-        connectedDb = node;
-        await languageClient.connect(connectedDb.treeItem.connectionString, connectedDb.treeItem.databaseName);
-        connectedDb.treeItem.isConnected = true;
-        await node.refresh();
+
+        const oldNode: IAzureNode | undefined = ext.connectedMongoDB;
+        await languageClient.connect(node.treeItem.connectionString, node.treeItem.databaseName);
         context.globalState.update(connectedDBKey, node.id);
+        ext.connectedMongoDB = node;
+        await tree.refresh(node.parent);
+
+        if (oldNode) {
+            await tree.refresh(oldNode.parent);
+        }
     });
     actionHandler.registerCommand('cosmosDB.deleteMongoDB', async (node?: IAzureNode<MongoDatabaseTreeItem>) => {
         if (!node) {
             node = <IAzureNode<MongoDatabaseTreeItem>>await tree.showNodePicker(MongoDatabaseTreeItem.contextValue);
         }
         await node.deleteNode();
-        if (connectedDb && connectedDb.treeItem.id === node.treeItem.id) {
-            connectedDb = null;
+        if (ext.connectedMongoDB && ext.connectedMongoDB.id === node.id) {
```

Comment:
Also need this in detach.  Although not sure there's anything bad that happens if it's not done?

Reply:
Filed a separate issue: https://github.com/Microsoft/vscode-cosmosdb/issues/492

Don't feel like fixing in this PR and it was always a problem before my changes